components/tdisplays3/t_display_s3.cpp

```diff
@@ -8,26 +8,132 @@ namespace tdisplays3 {
 
 static const char *const TAG = "TDisplayS3";
 
+#ifdef TDISPLAYS3_USE_ASYNC_IO
+namespace writer_shim {
+  static TaskHandle_t writeFrameTask_ = nullptr;
+  static int16_t x1 = 0;
+  static int16_t x2 = 0;
+  static int16_t y1 = 0;
+  static int16_t y2 = 0;
+  static uint16_t *buffer = nullptr;
+  static void (*callback)(void) = nullptr;
+
+  // A background task used to write the framebuffer to the display
+  static void IRAM_ATTR write_framebuffer_task(void *pv_params) {
+
+    auto tft = (TFT_eSPI *) pv_params;
+
+    while (true) {
+      ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+
+      if (x1 == 0 && y1 == 0 && x2 == 0 && y2 == 0) {
```

Comment:
Wouldn't a better condition be
`if ((x1 == 0 && x2 == 0) || (y1==0 && y2 == 0))`?
i.e. what happens if one of the dimensions is empty, but the other is not? (Corner case, but is it fully impossible?)

Reply:
Yes sir, your suggestion is the better implementation :)